### PR TITLE
Mes 2975 error messaging

### PR DESCRIPTION
**Pull request by ataylorBJSS**
## Description and relevant Jira numbers
Display generic error message with link for journal refresh and search errors.
https://jira.dvsacloud.uk/browse/MES-2975

Simulator Screen Shot - iPad Pro (11-inch) - 2019-07-30 at 16 02 06
Simulator Screen Shot - iPad Pro (11-inch) - 2019-07-30 at 16 01 15

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/components/error-message/error-message.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+@Component({
+  selector: 'error-message',
+  templateUrl: 'error-message.html',
+})
+export class ErrorMessageComponent {
+
+  @Input() returnTo: string;
+  @Input() showAdditionalText: boolean;
+
+  @Output() exitModal = new EventEmitter<string>();
```
Comment (johnmorleybjss) [inline src/components/error-message/error-message.ts]: Could be `EventEmitter<void>` if you never call `emit` with any argument.

File: src/pages/test-results-search/test-results-search.ts [hunk 2]
```diff
@@ -87,6 +92,12 @@ export class TestResultsSearchPage extends BasePageComponent {
             this.store$.dispatch(new SaveLog(log));
             this.searchResults = [];
             this.showSearchSpinner = false;
+
+            if (err) {
+              this.showError(err);
+              return of(this.hasSearched = false);
```
Comment (johnmorleybjss) [inline src/pages/test-results-search/test-results-search.ts]: Do you mean to do assignment yielding a value here? Or is it meant to be a comparison?

File: src/pages/view-test-result/__tests__/view-test-result.spec.ts [hunk 3]
```diff
@@ -376,7 +377,7 @@ describe('ViewTestResultPage', () => {
       component.showErrorMessage = true;
       fixture.detectChanges();
 
-      expect(fixture.debugElement.query(By.css('.error'))).not.toBeNull();
+      // expect(fixture.debugElement.query(By.css('.error'))).not.toBeNull();
```
Comment (johnmorleybjss) [inline src/pages/view-test-result/__tests__/view-test-result.spec.ts]: Commented.

**Review by ghost — APPROVED** [review]

File: src/pages/error-page/error.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  private errorType: string;
+  public errorLink: string;
+  public showAdditionalText: boolean;
+
+  constructor(public navController: NavController, public navParams: NavParams) { }
+
+  ngOnInit(): void {
+    this.errorType = this.navParams.get('type');
+    this.errorLink = ErrorTypes[this.errorType];
+    this.showAdditionalText = this.errorType === 'SEARCH';
```
Comment (Cavalier07) [inline src/pages/error-page/error.ts]: Can we use the enum here

File: src/pages/journal/journal.ts [hunk 5]
```diff
@@ -185,8 +187,15 @@ export class JournalPage extends BasePageComponent implements OnInit {
...
     if (error === undefined || error.message === '') return;
-    this.createToast(error.message);
-    this.toast.present();
+
+    // Modals are at the same level as the ion-nav so are not getting the zoom level class,
+    // this needs to be passed in the create options.
+    const zoomClass = `modal-fullscreen ${this.app.getTextZoomClass()}`;
+    const errorModal = this.modalController.create(
+      ERROR_PAGE,
+      { type: 'JOURNAL_REFRESH' },
```
Comment (Cavalier07) [inline src/pages/journal/journal.ts]: Should use enum

**Review by TomBillingtonUK — APPROVED** [review]
A couple of comments. Need screenshots from the app and also some unit tests before merging

File: src/pages/error-page/error.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import { ErrorTypes } from '../../shared/models/error-message';
+
+@IonicPage()
+@Component({
+  selector: 'page-error',
+  templateUrl: 'error.html',
+})
+export class ErrorPage {
+
+  private errorType: string;
```
Comment (Cavalier07) [inline src/pages/error-page/error.ts]: Can type this as the enum

File: src/pages/view-test-result/__tests__/view-test-result.spec.ts (hunk 3 above)
Comment (ataylorBJSS) [reply src/pages/view-test-result/__tests__/view-test-result.spec.ts]: The test has now been uncommented.

File: src/pages/error-page/error.ts (hunk 4 above)
Comment (ataylorBJSS) [reply src/pages/error-page/error.ts]: yeah, done.

File: src/pages/journal/journal.ts (hunk 5 above)
Comment (ataylorBJSS) [reply src/pages/journal/journal.ts]: definitely, updated!

File: src/pages/error-page/error.ts (hunk 6 above)
Comment (ataylorBJSS) [reply src/pages/error-page/error.ts]: updated in latest push.

File: src/components/error-message/error-message.ts (hunk 1 above)
Comment (ataylorBJSS) [reply src/components/error-message/error-message.ts]: Updated in latest push.

File: src/pages/test-results-search/test-results-search.ts (hunk 2 above)
Comment (ataylorBJSS) [reply src/pages/test-results-search/test-results-search.ts]: Yes, it's mean't to be an assignment. Moved to outside of return statement for clarity.

**Review by TomBillingtonUK — APPROVED** [review]

**Pull request merged**